### ISSUE TITLE
streaming upload

### ISSUE THREAD

**Issue by alekibango** [theelous3/asks#90]
if i read the code correctly,
it is not possible to make PUT request and stream data into it.

But that would be really nice to have.
Streaming by chunks of data or from file-like object is really a must for uploading  large files...

**Comment by theelous3** [comment]
You are correct and this is something that needs to be addressed quite quickly. I'll handle it this weekend.

**Comment by alekibango** [comment]
Thanks!

You would be surprized how many libraries ignore such corner cases.

Uploading large files using PUT (which is the only sane method) is painfull or impossible also on server side. They usually think large file is around 100MB.   Most pythonic web servers never try uploading file having size over 2 GB.

But i am uploading/accepting  at least tens of TB daily.... :) it really makes a difference to me.

**Comment by derfred** [comment]
Is there any update on this? I have a similar use-case.
